sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -371,16 +431,24 @@ class SessionCatalog(
    */
   def getTempViewOrPermanentTableMetadata(name: TableIdentifier): CatalogTable = synchronized {
```

Comment:
yea, I'd like to rename them to `getPersistedTableMetadataOption`, but probably not in this PR